spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java

```diff
@@ -162,4 +147,59 @@ public void serviceCreationFailed(Class<?> type, Class<?> impl, Throwable except
 		});
 		return locator.getService(RepositorySystem.class);
 	}
+
+	@Override
+	public Resource[] resolve(Coordinates root, Coordinates[] includes, CoordinatesFilter[] excludes) {
+		Assert.hasText(root.getGroupId(), "'groupId' cannot be blank.");
```

Comment:
We'll need to change these assertions message to clarify it pertains to `root`.

And iterate over the includes, and assert